thentos-core/src/Thentos/Types.hs

```diff
@@ -507,6 +509,22 @@ instance ToJSON ByUserOrServiceId where
     toJSON (ByService i k) = Aeson.object [ "service" .= Aeson.toJSON (i, k)]
 
 
+-- * send email
+
+-- TODO: Service ID? Pseudonyms? From address?
```

Comment:
Service ID is not required for now.

Pseudonyms: yes, that would be nice.  Do you need details on this?  Essentially, they are a services handle for a user, and need to be looked up, mapped to the user, and then lead to the email address.